test/http.js

```diff
@@ -115,6 +133,31 @@ test('response contains url', async t => {
 	t.is((await got(s.url)).url, `${s.url}/`);
 });
 
+test.serial('cancel in-progress request', async t => {
+	const body = new Readable({
+		read() {}
+	});
+	body.push(String(1));
+
+	const p = got(`${s.url}/no-res-store-id`, {body});
+
+	setImmediate(() => {
+		p.cancel();
+		body.push(null); // Complete the request that should now be canceled
+	});
+
+	await t.throws(p, PCancelable.CancelError);
+	await t.notThrows(() => cancelPromise);
```

Comment:
I just tried this. The test fails without. Turned it back into what it was. I remember now I wrote it your suggested way first. Not sure why this demands to be lazily evaluated. My JS-foo is not strong enough (yet).